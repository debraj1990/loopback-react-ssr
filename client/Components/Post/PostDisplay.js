import React from 'react'


const PostDetail = (props) => (
     <div>
        <div className="col-sm-4">
        <h1>{props.title}</h1>
        <p>{props.description}</p>
        </div>
    </div>
)

export default PostDetail