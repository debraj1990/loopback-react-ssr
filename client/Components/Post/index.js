import React, { Component } from 'react';
import axios from 'axios'
import Navbar from '../Navbar';
import PostDetail from './PostDisplay';
class Post extends Component {

    state = {
        posts: [],
        error: false
    }

    async componentDidMount() {

       try {
            const posts = (await axios.get('http://localhost:3000/api/Posts')).data
            this.setState({posts})
       } catch (e) {
           this.setState({error:true})
       }
       
    }

    renderPost() {
        return this.state.posts.map(post => {
            return <PostDetail key={post.id} {...post} />
        })
    }

    render(){
        const {error} = this.state
        return (
            <div className="container-fluid bg-3 text-center"> 
                <h3>Find Our Awesome Posts</h3>
                {error && 
                    <div className="alert alert-danger">
                    <strong>Sorry Your internet might have issues connecting!</strong>.
                  </div>
                }
                <div className="row">
                    {this.renderPost()}
                </div>
            </div>
        )
    }
}

export default Post