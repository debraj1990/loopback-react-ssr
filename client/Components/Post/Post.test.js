import React from 'react';
import {configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';
import PostDetail from './PostDisplay'

import Post from './'


configure({adapter: new Adapter });


describe('Login Component', () => {
  
  it('component should render the compoent without crushing', () => {
    const component = shallow(<Post />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  });

  

it('should not render any PostDetails when post is empty', () => {

  const posts = []
  const wrapper = shallow(<Post posts={posts} />)
  expect(wrapper.find(PostDetail)).toHaveLength(0)
 });
     
})