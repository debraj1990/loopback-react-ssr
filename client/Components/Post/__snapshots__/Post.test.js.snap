// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component component should render the compoent without crushing 1`] = `
<div
  className="container-fluid bg-3 text-center"
>
  <h3
    style={
      Object {
        "color": "blue",
      }
    }
  >
    Find Our Awesome Posts
  </h3>
  <div
    className="row"
  />
</div>
`;
